roachtest: django failed

[(roachtest).django failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2223072&tab=buildLog) on [master@ef55609797c92e46eb9efd08facc9db49558291d](https://github.com/cockroachdb/cockroach/commits/ef55609797c92e46eb9efd08facc9db49558291d):

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/django/run_1
	orm_helpers.go:218,orm_helpers.go:144,django.go:204,django.go:216,test_runner.go:754: 
		Tests run on Cockroach v20.2.0-alpha.1-2782-gef55609797
		Tests run against django cockroach-3.0.x
		8031 Total Tests Run
		8029 tests passed
		2 tests failed
		489 tests skipped
		21 tests ignored
		0 tests passed unexpectedly
		2 tests failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: expressions.tests.BasicExpressionsTests.test_case_in_filter_if_boolean_output_field (unexpected)
		--- FAIL: get_or_create.tests.UpdateOrCreateTransactionTests.test_creation_in_transaction (unexpected)
		--- SKIP: backends.mysql.test_creation.DatabaseCreationTests.test_create_test_db_unexpected_error due to MySQL tests (expected)
		For a full summary look at the django artifacts 
		
		An updated blocklist (djangoBlocklist20_2) is available in the artifacts' django log

```

<details><summary>More</summary><p>


Artifacts: [/django](https://teamcity.cockroachdb.com/viewLog.html?buildId=2223072&tab=artifacts#/django)
Related:
- #52761 roachtest: django failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-20.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-20.1) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Adjango.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).django failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2226493&tab=buildLog) on [master@fb25dbe53c2bc99e673c292b71d73d30247a7450](https://github.com/cockroachdb/cockroach/commits/fb25dbe53c2bc99e673c292b71d73d30247a7450):

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/django/run_1
	orm_helpers.go:218,orm_helpers.go:144,django.go:204,django.go:216,test_runner.go:754: 
		Tests run on Cockroach v20.2.0-alpha.1-2818-gfb25dbe53c
		Tests run against django cockroach-3.0.x
		8031 Total Tests Run
		8029 tests passed
		2 tests failed
		489 tests skipped
		21 tests ignored
		0 tests passed unexpectedly
		2 tests failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- SKIP: backends.mysql.test_creation.DatabaseCreationTests.test_create_test_db_unexpected_error due to MySQL tests (expected)
		--- FAIL: expressions.tests.BasicExpressionsTests.test_case_in_filter_if_boolean_output_field (unexpected)
		--- FAIL: get_or_create.tests.UpdateOrCreateTransactionTests.test_creation_in_transaction (unexpected)
		For a full summary look at the django artifacts 
		
		An updated blocklist (djangoBlocklist20_2) is available in the artifacts' django log

```

<details><summary>More</summary><p>


Artifacts: [/django](https://teamcity.cockroachdb.com/viewLog.html?buildId=2226493&tab=artifacts#/django)
Related:
- #53524 roachtest: django failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202008261913_v20.2.0-beta.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202008261913_v20.2.0-beta.1) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #52761 roachtest: django failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-20.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-20.1) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Adjango.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).django failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2229351&tab=buildLog) on [master@123ca0782f423bc3d225a5ca798ece60c3185935](https://github.com/cockroachdb/cockroach/commits/123ca0782f423bc3d225a5ca798ece60c3185935):

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/django/run_1
	orm_helpers.go:218,orm_helpers.go:144,django.go:204,django.go:216,test_runner.go:754: 
		Tests run on Cockroach v20.2.0-alpha.1-2874-g123ca0782f
		Tests run against django cockroach-3.0.x
		8032 Total Tests Run
		8030 tests passed
		2 tests failed
		488 tests skipped
		21 tests ignored
		0 tests passed unexpectedly
		2 tests failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: expressions.tests.BasicExpressionsTests.test_case_in_filter_if_boolean_output_field (unexpected)
		--- FAIL: get_or_create.tests.UpdateOrCreateTransactionTests.test_creation_in_transaction (unexpected)
		--- SKIP: backends.mysql.test_creation.DatabaseCreationTests.test_create_test_db_unexpected_error due to MySQL tests (expected)
		For a full summary look at the django artifacts 
		
		An updated blocklist (djangoBlocklist20_2) is available in the artifacts' django log

```

<details><summary>More</summary><p>


Artifacts: [/django](https://teamcity.cockroachdb.com/viewLog.html?buildId=2229351&tab=artifacts#/django)
Related:
- #53524 roachtest: django failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202008261913_v20.2.0-beta.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202008261913_v20.2.0-beta.1) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #52761 roachtest: django failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-20.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-20.1) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Adjango.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).django failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2231078&tab=buildLog) on [master@8b91062f9351d18f9104aff567cb152df162021e](https://github.com/cockroachdb/cockroach/commits/8b91062f9351d18f9104aff567cb152df162021e):

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/django/run_1
	orm_helpers.go:218,orm_helpers.go:144,django.go:204,django.go:216,test_runner.go:754: 
		Tests run on Cockroach v20.2.0-alpha.1-2896-g8b91062f93
		Tests run against django cockroach-3.0.x
		8031 Total Tests Run
		8030 tests passed
		1 test failed
		489 tests skipped
		21 tests ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: get_or_create.tests.UpdateOrCreateTransactionTests.test_creation_in_transaction (unexpected)
		--- SKIP: backends.mysql.test_creation.DatabaseCreationTests.test_create_test_db_unexpected_error due to MySQL tests (expected)
		For a full summary look at the django artifacts 
		
		An updated blocklist (djangoBlocklist20_2) is available in the artifacts' django log

```

<details><summary>More</summary><p>


Artifacts: [/django](https://teamcity.cockroachdb.com/viewLog.html?buildId=2231078&tab=artifacts#/django)
Related:
- #53524 roachtest: django failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202008261913_v20.2.0-beta.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202008261913_v20.2.0-beta.1) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #52761 roachtest: django failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-20.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-20.1) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Adjango.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).django failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2232292&tab=buildLog) on [master@8746e7e347dbbc19be051168225cd4fab0582294](https://github.com/cockroachdb/cockroach/commits/8746e7e347dbbc19be051168225cd4fab0582294):

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/django/run_1
	orm_helpers.go:225,orm_helpers.go:151,django.go:204,django.go:216,test_runner.go:754: 
		Tests run on Cockroach v20.2.0-alpha.1-2900-g8746e7e347
		Tests run against django cockroach-3.0.x
		8032 Total Tests Run
		8029 tests passed
		3 tests failed
		488 tests skipped
		21 tests ignored
		0 tests passed unexpectedly
		3 tests failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: admin_views.tests.AdminViewBasicTest.test_change_list_sorting_model_admin_reverse (unexpected)
		--- SKIP: backends.mysql.test_creation.DatabaseCreationTests.test_create_test_db_unexpected_error due to MySQL tests (expected)
		--- FAIL: expressions.tests.BasicExpressionsTests.test_case_in_filter_if_boolean_output_field (unexpected)
		--- FAIL: get_or_create.tests.UpdateOrCreateTransactionTests.test_creation_in_transaction (unexpected)
		For a full summary look at the django artifacts 
		
		An updated blocklist (djangoBlocklist20_2) is available in the artifacts' django log

```

<details><summary>More</summary><p>


Artifacts: [/django](https://teamcity.cockroachdb.com/viewLog.html?buildId=2232292&tab=artifacts#/django)
Related:
- #53524 roachtest: django failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202008261913_v20.2.0-beta.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202008261913_v20.2.0-beta.1) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #52761 roachtest: django failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-20.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-20.1) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Adjango.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).django failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2233513&tab=buildLog) on [master@cd8267ca84f6a0267a898220cfaa6630188b45c5](https://github.com/cockroachdb/cockroach/commits/cd8267ca84f6a0267a898220cfaa6630188b45c5):

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/django/run_1
	orm_helpers.go:225,orm_helpers.go:151,django.go:204,django.go:216,test_runner.go:754: 
		Tests run on Cockroach v20.2.0-alpha.1-2902-gcd8267ca84
		Tests run against django cockroach-3.0.x
		8031 Total Tests Run
		8030 tests passed
		1 test failed
		489 tests skipped
		21 tests ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: get_or_create.tests.UpdateOrCreateTransactionTests.test_creation_in_transaction (unexpected)
		--- SKIP: backends.mysql.test_creation.DatabaseCreationTests.test_create_test_db_unexpected_error due to MySQL tests (expected)
		For a full summary look at the django artifacts 
		
		An updated blocklist (djangoBlocklist20_2) is available in the artifacts' django log

```

<details><summary>More</summary><p>


Artifacts: [/django](https://teamcity.cockroachdb.com/viewLog.html?buildId=2233513&tab=artifacts#/django)
Related:
- #53524 roachtest: django failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202008261913_v20.2.0-beta.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202008261913_v20.2.0-beta.1) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #52761 roachtest: django failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-20.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-20.1) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Adjango.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/cmd/roachtest/django.go
+++ b/pkg/cmd/roachtest/django.go
@@ -61,15 +61,15 @@ func registerDjango(r *testRegistry) {
 			c,
 			node,
 			"install dependencies",
-			`sudo apt-get -qq install make python3.6 libpq-dev python3.6-dev gcc python3-setuptools python-setuptools build-essential`,
+			`sudo apt-get -qq install make python3.7 libpq-dev python3.7-dev gcc python3-setuptools python-setuptools build-essential`,
 		); err != nil {
 			t.Fatal(err)
 		}
 
 		if err := repeatRunE(
-			ctx, c, node, "set python3.6 as default", `
+			ctx, c, node, "set python3.7 as default", `
     		sudo update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.5 1
-    		sudo update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.6 2
+    		sudo update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.7 2
     		sudo update-alternatives --config python3`,
 		); err != nil {
 			t.Fatal(err)
@@ -77,7 +77,7 @@ func registerDjango(r *testRegistry) {
 
 		if err := repeatRunE(
 			ctx, c, node, "install pip",
-			`curl https://bootstrap.pypa.io/get-pip.py | sudo -H python3.6`,
+			`curl https://bootstrap.pypa.io/get-pip.py | sudo -H python3.7`,
 		); err != nil {
 			t.Fatal(err)
 		}

